ginga/rv/plugins/WBrowser.py

```diff
@@ -0,0 +1,177 @@
+                ('Reload page', lambda w: self.reload_cb(), 'rotate_48.png'),
+                ('Stop loading', lambda w: self.stop_cb(), 'stop_48.png'),
+                ('Go to top of documentation', lambda w: self.show_help(),
+                 'fits.png')):
+            btn = tbar.add_action(
+                None, iconpath=os.path.join(self.fv.iconpath, ico))
+            btn.add_callback('activated', cb)
+            btn.set_tooltip(tt)
+        container.add_widget(tbar, stretch=0)
+
+        btns = Widgets.HBox()
+        btns.set_border_width(4)
+        btns.set_spacing(3)
+        btn = Widgets.Button('Close')
+        btn.add_callback('activated', lambda w: self.close())
+        btns.add_widget(btn, stretch=0)
+        btn = Widgets.Button('Help')
+        btn.add_callback('activated', lambda w: self.help())
+        btns.add_widget(btn, stretch=0)
+        btns.add_widget(Widgets.Label(''), stretch=1)
+        container.add_widget(btns, stretch=0)
+
+        self.gui_up = True
+
+    def _download_doc(self, plugin=None, no_url_callback=None):
+        from ginga.doc.download_doc import get_doc
+        self.fv.assert_nongui_thread()
+
+        self.fv.gui_do(self._load_doc, WAIT_HTML, url_is_content=True)
+
+        def _dl_indicator(count, blksize, totalsize):
+            pct = (count * blksize) / totalsize
+            msg = 'Downloading: {:.1%} complete'.format(pct)
+            self.fv.gui_do(self.entry.set_text, msg)
+
+        # This can block as long as it takes without blocking the UI.
+        url = get_doc(logger=self.logger, plugin=plugin,
+                      reporthook=_dl_indicator)
+
+        self.fv.gui_do(self._load_doc, url, no_url_callback=no_url_callback)
+
+    def _load_doc(self, url, no_url_callback=None, url_is_content=False):
+        self.fv.assert_gui_thread()
+        if url is None:
+            self.entry.set_text('')
+            if no_url_callback is None:
+                self.fv.show_error("Couldn't load web page")
+            else:
+                no_url_callback()
+        else:
+            self.browse(url, url_is_content=url_is_content)
+
+    def show_help(self, plugin=None, no_url_callback=None):
+        """See `~ginga.GingaPlugin` for usage of optional keywords."""
+        if not Widgets.has_webkit:
+            return
+
+        self.fv.nongui_do(self._download_doc, plugin=plugin,
+                          no_url_callback=no_url_callback)
+
+    def browse(self, url, url_is_content=False):
+        if not Widgets.has_webkit:
+            return
+
+        try:
+            if url_is_content:  # This was load_html()
+                self.browser.load_html_string(url)
+                self.entry.set_text('')
+            else:
+                self.logger.debug("Browsing '{}'".format(url))
+                self.browser.load_url(url)
+                self.entry.set_text(url)
+        except Exception as e:
+            self.fv.show_error("Couldn't load web page: {}".format(str(e)))
```

Comment:
> The best scenario would be to try to display online first (ideally for the matching version, but that might be asking too much) and if that fails, to fall back to the docstring (in some way, whether rendered by the browser or simple text widget). I don't know how easy it is to detect a failure with the WebKit browser.

Does a WebKit load failure go here? If so, it might be possible to replace `show_error` with some kind of docstring display. But if not, then there is no point modifying this.